eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -423,27 +423,47 @@ object Graph {
      * Representation of an edge of the graph
      *
      * @param desc        channel description
-     * @param update      channel info
+     * @param source      source of the channel parameters: can be a channel_update or hints from an invoice // TODO: should we do a class hierarchy instead: KnownEdge/AssistedEdge?
```

Comment:
No, I think it's better if the source details don't leak. I don't see any benefit of also splitting GraphEdge in KnownEdge/AssistedEdge.

Reply:
Ok, btw my plan is to eventually get rid of `ChannelDesc` and replace it by a `case class EdgeId(a: PublicKey, b: PublicKey)` like you suggested.